src/formatPkg.js

```diff
@@ -431,3 +433,18 @@ function getAlternativeNames(name) {
     new Set([concatenatedName, splitName, dotJSName, normalName])
   );
 }
+
+function getModuleType(cleaned) {
+  const main = cleaned.main || '';
+  if (
+    typeof cleaned.module === 'string' ||
+    cleaned.type === 'module' ||
+    main.endsWith('.mjs')
+  ) {
+    return 'esm';
+  }
+  if (cleaned.type === 'commonjs' || main.endsWith('.cjs')) {
+    return 'cjs';
+  }
+  return 'unknown';
```

Comment:
That is good iteration to begin with. Like I said it's may be a bit too simplistic but it can be improved later.